### PR TITLE
feature/#13650-veritone-endpointveritone-user-improvements

### PR DESCRIPTION
**Pull request by angel-perez-itti**
Pull request for the topic https://github.com/slingr-stack/platform/issues/13650 created by @pasaperez-slingr 

## What does it do?

- The endpoint is updated to be able to use the different development environments and geographic regions offered by Veritone.
- Added support for 3 Veritone Rest APIs.
- Documentation errors are corrected.
- Fixed bugs for authentication and endpoint usage. 
- Added an extra flow step to be able to make calls to all endpoints. 

## How to test it?

Make calls from the platform for example using the console using the information provided in the documentation. **Use the v1.0.33-dev for testing this PR**

## Implementation notes

Create a new tags changing the major version of the endpoint, it's partially backward compatible. Remove previous versions with "-dev" postfix.

### CODE REVIEW

**Comment by sebasmoyano** [comment]
#approve

File: endpoint.json [hunk 1]
```diff
@@ -25,24 +25,55 @@
...
+            "description": "Sends a OPTIONS request"
         }
     ],
     "scripts": [
         "functions.js"
     ],
     "flowSteps": [
-        "apiCallVeritoneuser"
+        "apiCallVeritoneuser",
+        "apiDataVeritoneuser"
```
Comment (agreggio-slingr) [inline endpoint.json]: If we are using camel case, I think it should be apiCallVeritoneUser and apiDataVeritoneUser

File: flowSteps/apiDataVeritoneuser/step.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			return endpoint._connect(options);
+		case 'head':
+			return endpoint._head(options);
+		case 'options':
+			return endpoint._options(options);
+		case 'patch':
+			return endpoint._patch(options);
+		case 'trace':
+			return endpoint._trace(options);
+	}
```
Comment (agreggio-slingr) [inline flowSteps/apiDataVeritoneuser/step.js]: On line 17, post is set by default, so the switch is not necessary. We can replace it with return endpoint._post(options);

File: src/main/resources/endpoint.properties [hunk 3]
```diff
@@ -15,4 +15,4 @@ _endpoints_services_api=http://localhost:2233/api
 _token=test1/dev/veritoneuser
 
 # Endpoint specific properties
-_endpoint_config={"environment":"stage", "clientId":"id", "clientSecret":"secret"}
+_endpoint_config={"environment":"stage", "region":"us-1", "clientId":"37d8b00c-9547-4b92-b6b6-f037c280b3c8", "clientSecret":"9kWCQoSyXrlzFfgPrE3VR80DYwr4sdhPWcOt7kfpFsxUvmbL3c-jJw", "redirectUri": "http://test1.localhost:8000/dev/endpoints/veritoneuser/authCallback"}
```
Comment (agreggio-slingr) [inline src/main/resources/endpoint.properties]: I would delete the default configurations

File: src/test/resources/test.properties [hunk 4]
```diff
@@ -16,5 +16,4 @@ _token=test1/dev/veritoneuser
 _testing_mode=true
 
 # Endpoint specific properties
-_endpoint_config={"environment":"production"}
-
+_endpoint_config={"environment":"stage", "region":"us-1", "clientId":"37d8b00c-9547-4b92-b6b6-f037c280b3c8", "clientSecret":"9kWCQoSyXrlzFfgPrE3VR80DYwr4sdhPWcOt7kfpFsxUvmbL3c-jJw", "redirectUri": "http://test1.localhost:8000/dev/endpoints/veritoneuser/authCallback"}
```
Comment (agreggio-slingr) [inline src/test/resources/test.properties]: Add it to the gitignore.

File: endpoint.json (hunk 1 above)
Comment (pasaperez-work) [reply endpoint.json]: I agree, it was changed. Just to make it clear the helper generator uses the name of the endpoint, in which it uses all its name in lowercase.

File: flowSteps/apiDataVeritoneuser/step.js (hunk 2 above)
Comment (pasaperez-work) [reply flowSteps/apiDataVeritoneuser/step.js]: I agree, it was simplified.

File: src/test/resources/test.properties (hunk 4 above)
Comment (pasaperez-work) [reply src/test/resources/test.properties]: It is only a token for documentation purposes. For when you want to use the alphanumeric format and length of the token to be clear, it is not a valid token.

File: endpoint.json (hunk 1 above)
Comment (agreggio-slingr) [reply endpoint.json]: I think the helpgen tool doesn't create steps. Looking at other already standardized endpoints, I see that we are using camelCase.

Screenshot from 2023-04-11 10-59-33

**Review by agreggio-slingr — APPROVED** [review]

File: endpoint.json (hunk 1 above)
Comment (pasaperez-work) [reply endpoint.json]: Exactly, it uses camel case, but the name of the endpoint actually (according to the endpoint.json) is "veritoneuser" all in lowercase as a single string, unlike the example that uses "github-app" also all in lowercase but divided with the middle hyphen as 2 words. 
So no camelCase algorithm could automatically identify the complete lowercase string.
Only one case that may be common for legacy endpoints.

**Comment by pasaperez-work** [comment]
Version v1.0.34 taged as new release.

**Pull request merged**